full-service/src/db/txo.rs

```diff
@@ -72,6 +72,14 @@ pub enum TxoStatus {
     Unverified,
 }
 
+#[derive(Debug, PartialEq)]
+pub enum RTHMemoType {
+    AuthenticatedSender,
+    AuthenticatedSenderWithPaymentIntentId,
+    AuthenticatedSenderWithPaymentRequestId,
+    Unsupported,
+}
```

Comment:
This should also maybe be just a general Memo enum, not Memo Type, and include the data in the actual enum that goes along with each of them

Reply:
```suggestion
pub enum Memo {
    AuthenticatedSender { address_hash: String },
    AuthenticatedSenderWithPaymentIntentId { address_hash: String, payment_intent_id: String },
    AuthenticatedSenderWithPaymentRequestId { address_hash: String, payment_request_id: String },
    ...remaining memos after this
}
```

Reply:
Actually, it looks like it can only be 1 to 1 relationship, so maybe at the database level it's broken into memo type and underlying data, and at the model level it combines them into a single enum that takes in values